src/graphics/spritebatch.rs

```diff
@@ -51,29 +54,64 @@ impl SpriteBatch {
     /// Adds a new sprite to the sprite batch.
     ///
     /// Returns a handle with which to modify the sprite using `set()`
-    pub fn add(&mut self, param: graphics::DrawParam) -> SpriteIdx {
-        self.sprites.push(param);
+    pub fn add(
+        &mut self, 
+        param: graphics::DrawParam
+    ) -> SpriteIdx {
+        let src_width = param.src.w;
+        let src_height = param.src.h;
+        // We have to mess with the scale to make everything
+        // be its-unit-size-in-pixels.
+        // We also invert the Y scale if our screen coordinates
+        // are "upside down", because by default we present the
+        // illusion that the screen is addressed in pixels.
+        // BUGGO: Which I rather regret now.
+        // let invert_y = if gfx.screen_rect.h < 0.0 { 1.0 } else { -1.0 };
+        // TODO: Figure out whether implementing this is needed/how to do it cleanly
```

Comment:
I think implementing a method on Image called, say, `convert_param()` that would do this for you might be helpful... problem is I don't want to make you pass in the context to the `add()` method but I'm not sure what a good way would be otherwise to get access to `gfx_context.screen_rect`